### ISSUE TITLE
Allow fitting the Young's modulus to `TensileTest` data

### ISSUE THREAD

**Issue by aplowman** [LightForm-group/formable#12]
We would need to modify `TensileTest` in the following ways:

- Add an `elastic_range` attribute (analogous to the current `plastic_range` attribute). Sensible default range would perhaps be [0, 0.01]
- Remove the default value of the Young's modulus - this doesn't make much sense. Instead, we should raise an exception when the Young's modulus is not set but a method that requires it is invoked.
- Add a method `fit_youngs_modulus` which should fit the true stress-strain data within the `elastic_range` to a straight line model and then set the `youngs_modulus` attribute (overwriting the previous value)
- Add more visualisations to the `TensileTest.show` method:
    - Elastic and plastic ranges
    - Fitted elastic data

**Comment by JQFonseca** [comment]
Why are we considering the elastic modulus in the formable package? Shouldn't it be restricted to the plastic part?

**Comment by aplowman** [comment]
The Young's modulus is [used ](https://github.com/LightForm-group/formable/blob/b2236fa3dfda7b293768d79e5a36c2d5b6919cc0/formable/tensile_test.py#L164)to find the plastic strain. 

I agree `formable` probably isn't the best place for the `TensileTest` class, given that it obviously has utility outside of formability analysis. We have a separate dedicated package [called tensile_test](https://github.com/LightForm-group/tensile-test) for this class. But I had moved the class into formable to make updates on the CSF easier whilst I was developing the Levenberg-Marquardt fitting functions.

We can move it back to the `tensile_test` package if you prefer.

**Comment by JQFonseca** [comment]
Apologies, I should have looked at the code better. The elastic part of the strain in experimental stress-strain data is often not the Young's modulus (the material property). It can contain compliance, for example. But if all you are doing is doing a linear estimate of the non-plastic strain then it's fine. Maybe call it compliance constant instead of modules, so it's clear it might not be the Young's modulus?

**Comment by aplowman** [comment]
>The elastic part of the strain in experimental stress-strain data is often not the Young's modulus (the material property). It can contain compliance, for example.

Ah I see. Would an example of this be compliance due to deformation of the testing machine? If not, could you give an example?

If we opted for `compliance_constant`, this would be the inverse of the current `youngs_modulus`?

**Comment by JQFonseca** [comment]
Yes that's it. For larger deformations we often use compression testing
where the samples are too small for on-sample extensometry
(compressometry?) With tensile testing extensometry is more common but it
is limited to smaller strains. Wayne has a good comparison of DIC tensile
testing and cross-head data.

And yes, I guess compliance is the inverse. Maybe we can use
stiffness_constant?

João

**Comment by aplowman** [comment]
OK, thanks! Let's go with the new method being `fit_stiffness_constant`, setting an attribute `stiffness_constant`. The `TensileTest.__init__` parameter `youngs_modulus` should also be changed to `stiffness_constant` for consistency, with a note added to the docstring that this is where the Young's modulus should be specified, if it is given.
